### ISSUE TITLE
ONNX exporter related bug

### ISSUE THREAD

**Issue by vuksan314** [pytorch/pytorch#46156]
## 🐛 Bug

torch.onnx.export produces err message: 0 INTERNAL ASSERT FAILED at "..\\torch\\csrc\\jit\\ir\\ir.cpp":515, please report a bug to PyTorch. 131 not in scope

## To Reproduce

Run the code provided

import torch
import utilities
import onnxruntime
import torch.nn as nn
import numpy as np
import onnx

@torch.jit.script
def fun1(arg1, arg2, arg3:int):
    device = arg1.device
    checkvar = arg1.clone().detach()
    arg2_result = arg2.clone().detach()
    boolvar = torch.zeros(1, dtype=torch.bool, device=device)

    boolvar[0] = False
    for idx in range(int(arg2.item())):
        if (checkvar[idx * 7 + arg3 + 1] == 0):
            boolvar[0] = True
            break
    if boolvar.item() is False:

        checkvar[arg2_result * 5 + arg3] = 356

        arg2_result = arg2_result + 1

    return checkvar, arg2_result

@torch.jit.script
def fun1_batch(var3, arg2, var13:int):

    tvar1 = var3.clone().detach()
    tvar2 = arg2.clone().detach()
    for b_idx in range(tvar1.shape[1]):
        tvar1[:, b_idx], tvar2[b_idx] = fun1(tvar1[:, b_idx], tvar2[b_idx], var13)
    return tvar1, tvar2

class dummyclass(nn.Module):

    def forward(self, input0, input1, input2, input3):

        device = input0.device
        arg2 = input2.clone().detach()
        var3 = input0.clone().detach()
        var3, arg2 = fun1_batch(var3, arg2, 1)

        return input0, input1

print(torch.__version__)

device = torch.device('cpu')

list_var = []

list_var.append(torch.ones( (53,1), dtype=torch.int32, device = device))
list_var.append(torch.ones( (18,1,4), device = device))
list_var.append(torch.ones( (1), dtype=torch.int64, device = device))
list_var.append(torch.ones( (14,1), dtype=torch.int32, device = device))

classvar = dummyclass() 
result = classvar(list_var[0], list_var[1], list_var[2], list_var[3])

torch.onnx.export(classvar, ( list_var[0], list_var[1], list_var[2], list_var[3]), "bugrepro.onnx",
                example_outputs=result,
                opset_version=12,
                verbose=True)

onnx_model = onnx.load("bugrepro.onnx")
problems = onnx.checker.check_model(onnx_model)
print("Exported model has been tested with ONNXRuntime, and the result looks good! As you can see probelms are: " + str(problems))
ort_session = onnxruntime.InferenceSession("bugrepro.onnx")
print("All good here")

## Expected behavior
Expected no error in producing ONNX
## Environment

PyTorch version: 1.8.0a0+8a1e100
Is debug build: True
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.18.2

Python version: 3.6 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.18.2
[pip3] torch==1.4.0+cpu
[pip3] torchvision==0.5.0+cpu
[conda] Could not collect

**Comment by spandantiwari** [comment]
@vuksan314 - there are lot of tensor detach and device() calls within your model. They may actually be creating issues here. If they are not required for model inference, can you please try removing them before ONNX export?

It may also be helpful to try using `torch.jit.script()` API to at least see if this model can be converted to a ScriptModule correctly.

Note that TorchScript and ONNX support a subset of PyTorch and there may be some changes to the model code that may be required. You can look at the doc page for some details:  https://pytorch.org/docs/master/onnx.html
